Hint at how to properly close issues with pull requests

JulianKniephoff opened the pull request:
Mentioning issues in your PR with some magic keywords can automatically close these issues when the PR is merged. We even hint at this in our PR-template already.

However, GitHubs implementation of this does not work super well with our release/branching model. Specifically, it only works if the PR is merged into the default branch, which is `develop` in our case.

It **can** be made to work in our situation if the magic phrases are mentioned in commit messages alternatively or in addition to the PR description, because we regularly forward merge the release branches.

This patch adds a corresponding note about this to the documentation,
and links it in the PR template.

lkiesow commented:
The information is already in the linked documentation. I'm not sure trying to turn the checklist into additional documentation (especially since this is copied into every pull request) is a good idea. If you want it in here, maybe just turn it into a very short instruction of what to do instead? Something like:

* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists. E.g. put `fixes #123` in the commit message

If you think a longer explanation is required, I suggest putting that one into the developer documentation and linking it from here instead of linking to the GitHub documentation? That allows you to properly explain what's going on and how this works given Opencast's branching model.

ziegenberg commented:
I second @lkiesow opinion.

JulianKniephoff commented:
IIRC that was the tenor in the technical as well. I got it, I will do it, but it's rather low on my list rn. ;D

mliradelc commented:
I Like @JulianKniephoff description, but also Lars link suggestion to keep it as a link for tidiness. The description on the GitHub page is not as clear as Julian's. 

Could it be possible to add Julian's explanation to the documentation and make a link to the GitHub PR template?

JulianKniephoff committed:
Hint at how to properly close issues with pull requests

Mentioning issues in your PR with some magic keywords can
automatically close these issues when the PR is merged. We even hint
at this in our PR-template already.

However, GitHubs implementation of this does not work super well with
our release/branching model. Specifically, it only works if the PR is
merged into the default branch, which is develop in our case.

It can be made to work in our situation if the magic phrases are
mentioned in commit messages alternatively or in addition to the PR
description, because we regularly forward merge the release branches.

This patch adds a corresponding note about this to the documentation,
and links it in the PR template.

JulianKniephoff commented:
Finally got to this again. (*Tries to get stuff off their plate.)

(Re-)requesting review from everyone who complained. xP

gregorydlogan reviewed (approved):
LGTM

JulianKniephoff commented:
Fixed the conflicts, which required me to move the blurb somewhere else. I hope this is fine. To not drag this out any further I will just merge it now without another round of reviews. It has a review, others had the chance, and this is basically a docs PR anyway. 🤷‍♀️
